### PR TITLE
Adding undefined and unknown values

### LINKED ISSUE #229: Introduce 'unknown' for event Direction
Issue name: “Introduce an 'unknown' for event Direction”
 ## Summary
Not all existing permit systems provide a direction. There will be a number of events where the direction will be unknown and difficult to infer based on the data. Adding an unknown rather than an 'educated guess' is likely to be better for the data consumer.
 ## Motivation
At the moment a direction HAS to be included - even if it is wrong. By being explicit where the direction is unknown the data consumer can make their own assessment and truly trust the data where the direction is defined.
 ## Proposed changes
Introduce a new option of 'Unknown'

### PR DESCRIPTION
**Pull request by jlarsonOmahaNE**
This PR resolves #229 by adding `undefined` and `unknown` to the [Direction](https://github.com/usdot-jpo-ode/wzdx/blob/main/spec-content/enumerated-types/Direction.md) enumerated types of the [RoadEventCoreDetail](https://github.com/usdot-jpo-ode/wzdx/blob/main/spec-content/objects/RoadEventCoreDetails.md#roadeventcoredetails-object)s to allow for no road direction or if the producer does not know affected direction of travel.

### CODE REVIEW

File: spec-content/enumerated-types/Direction.md [hunk 1]
```diff
@@ -8,6 +8,8 @@ Value | Description
 `eastbound` | Road flow is in the eastbound direction
 `southbound` | Road flow is in the southbound direction
 `westbound` | Road flow is in the westbound direction
+`undefined` | Used if a data producer has start and end location for a road, but the road doesn't have a direction. Allows data consumers to assume that first coordinate is the start of an actual work zone.
```
Comment (mark-mockett) [inline spec-content/enumerated-types/Direction.md:11]: I'd rephrase this as something like "Road flow does not have a signed direction, but the beginning and end points of the corresponding RoadEventFeature represent the begin and end of the event in the direction of travel it impacts."

File: spec-content/enumerated-types/Direction.md [hunk 2]
```diff
@@ -8,6 +8,8 @@ Value | Description
...
+`unknown` | 	Used if a data producer doesn't know for sure which direction of travel is affected (including on a roadway with signed directions). Create an event for each direction and use the `unknown` enumeration. The data consumer cannot safely assume that work is affecting this direction of travel.
```
Comment (mark-mockett) [inline spec-content/enumerated-types/Direction.md:12]: similar to comment on `undefined`, I'd rephrase to match the others (i.e. start with "Road flow is...") and move the "Create an event" sentence to a guidance section, but reference it here.

**Review by mark-mockett — COMMENTED** [review]
@jlarsonOmahaNE I added a couple comments about suggested changes to the `unknown` and `undefined` directions to fit with the other enumerations. Also, the v4.1 schema still needs to be updated to include these new values - let me know if you need any help/instructions on how to do that.

File: spec-content/enumerated-types/Direction.md [hunk 3]
```diff
@@ -8,6 +8,8 @@ Value | Description
 `eastbound` | Road flow is in the eastbound direction
 `southbound` | Road flow is in the southbound direction
 `westbound` | Road flow is in the westbound direction
+`undefined` | Road flow does not have a signed direction, but the beginning and end points of the corresponding [RoadEventFeature](/spec-content/objects/RoadEventFeature.md) represent the begin and end of the event in the direction of travel it impacts.
```
Comment (mark-mockett) [inline spec-content/enumerated-types/Direction.md:11]: Add DeviceFeature note

File: spec-content/enumerated-types/Direction.md [hunk 4]
```diff
@@ -8,6 +8,8 @@ Value | Description
...
+`unknown` | 	Road flow may have a signed direction, but the affected direction of travel is not known. See guidance.
```
Comment (mark-mockett) [inline spec-content/enumerated-types/Direction.md:12]: Remove "see guidance"

File: spec-content/enumerated-types/Direction.md [hunk 5]
```diff
@@ -8,6 +8,8 @@ Value | Description
 `eastbound` | Road flow is in the eastbound direction
 `southbound` | Road flow is in the southbound direction
 `westbound` | Road flow is in the westbound direction
+`undefined` | Road flow does not have a signed direction. The beginning and end points of the corresponding [RoadEventFeature](/spec-content/objects/RoadEventFeature.md) represent the begin and end of the event in the direction of travel it impacts. Does not apply to DeviceFeatures.
```
Comment (j-d-b) [inline spec-content/enumerated-types/Direction.md:11]: I'd recommend rephrasing the second and third sentences to:

> For a RoadEventFeature, the first and last coordinates in the feature's geometry represent the begin and end of the road event in the direction of travel it impacts.

This provides more clarity than "begin and end points".

Also, removing the "Does not apply to DeviceFeatures" is helpful because the "undefined" type could be used/does apply to field devices, just not the part about the begin and end coordinates.

Comment (jlarsonOmahaNE) [reply spec-content/enumerated-types/Direction.md:11]: @j-d-b Cool thanks, Should the RoadEventFeature link also stay?

File: spec-content/enumerated-types/Direction.md [hunk 6]
```diff
@@ -8,7 +8,7 @@ Value | Description
 `eastbound` | Road flow is in the eastbound direction
 `southbound` | Road flow is in the southbound direction
 `westbound` | Road flow is in the westbound direction
-`undefined` | Road flow does not have a signed direction. The beginning and end points of the corresponding [RoadEventFeature](/spec-content/objects/RoadEventFeature.md) represent the begin and end of the event in the direction of travel it impacts. Does not apply to DeviceFeatures.
+`undefined` | Road flow does not have a signed direction. For a [RoadEventFeature](/spec-content/objects/RoadEventFeature.md), the first and last coordinates in the feature's geometry represent the begininning and end of the road event in the direction of travel it impacts.
```
Comment (j-d-b) [inline spec-content/enumerated-types/Direction.md:11]: Just a typo: "begininning" -> "beginning", otherwise looks good.

Comment (jlarsonOmahaNE) [reply spec-content/enumerated-types/Direction.md]: Haha thanks, totally copied in some extra letters

**Review by j-d-b — APPROVED** [review]

**Review by sknick-iastate — APPROVED** [review]

**Review by mark-mockett — APPROVED** [review]
